nsqadmin/static/js/views/channel.js

```diff
@@ -21,11 +21,18 @@ var ChannelView = BaseView.extend({
     initialize: function() {
         BaseView.prototype.initialize.apply(this, arguments);
         this.listenTo(AppState, 'change:graph_interval', this.render);
-        this.model.fetch().done(function() {
-            this.template = require('./channel.hbs');
-            this.render();
-            Pubsub.trigger('view:ready');
-        }.bind(this));
+        this.model.fetch()
+            .done(function(data) {
+                    this.template = require('./channel.hbs');
+                    this.render({'message': data['message']});
+                }.bind(this))
+            .fail(function(jqXHR) {
+                    this.template = require('./error.hbs');
+                    this.handleViewError(jqXHR);
+                }.bind(this))
+            .always(function() {
+                    Pubsub.trigger('view:ready');
+                });
```

Comment:
If you handled the templates like I was suggesting, you could do something like this:

```
        this.model.fetch()
            .done(function(data) {
                this.render({'message': data['message']});
            }.bind(this))
            .fail(this.handleViewError.bind(this))
            .always(Pubsub.trigger.bind(Pubsub, 'view:ready'));
```

But you'd need to make sure `template: require('./channel.hbs')` was in your View definition.

If you didn't mind `data`'s other properties getting into the render method, you could simplify further:

```
        this.model.fetch()
            .done(this.render.bind(this))
            .fail(this.handleViewError.bind(this))
            .always(Pubsub.trigger.bind(Pubsub, 'view:ready'));
```
...though it's not super explicit that `render` and `handleViewError` are both receiving arguments as the `done` and `fail` callbacks.